lib/redux/middleware/sentry_api_middleware.dart

```diff
@@ -15,49 +16,43 @@ class SentryApiMiddleware extends MiddlewareClass<AppState> {
     if (action is FetchOrganizationsAndProjectsAction) {
+      store.dispatch(LoadingAction(true));
       final thunkAction = (Store<AppState> store) async {
         final api = SentryApi(store.state.globalState.authToken);
         try {
+          store.dispatch(LoadingAction(true, 'Fetching Organization...'));
           final organizations = await api.organizations();
           final individualOrganizations = <Organization>[];
-          final Map<String, Cursor> projectCursorsByOrganizationSlug = {};
-          final Map<String, List<Project>> projectsByOrganizationId = {};
           final Set<String> projectIdsWithSessions = {};
+          final Map<String, List<Project>> projectsByOrganizationId = {};
+
+          var fullProgress = organizations.length * 2;
+          var currentProgress = 0;
 
           for (final organization in organizations) {
             final individualOrganization = await api.organization(organization.slug);
             individualOrganizations.add(individualOrganization ?? organization);
-            final currentCursor = store.state.globalState.projectCursorsByOrganizationSlug != null && !action.reload
-                ? store.state.globalState.projectCursorsByOrganizationSlug[organization.slug]
-                : null;
 
-            final nextCursor = action.pagination
-                ? currentCursor == null
-                ? Cursor(10, 0, 0)
-                : Cursor(10, currentCursor.offset + 1, 0)
-                : null;
-
-            final projects = await api.projects(organization.slug, nextCursor);
+            store.dispatch(LoadingAction(true, '${organization.name} -> Fetching projects...', ++currentProgress / fullProgress));
+            final projects = await api.projects(organization.slug);
             if (projects.isNotEmpty) {
               projectsByOrganizationId[organization.slug] = projects;
             }
 
-            final projectsWithSessionsForOrganization = await api.projectIdsWithSessions(organization.slug);
-            projectIdsWithSessions.addAll(projectsWithSessionsForOrganization);
-
-            if (action.pagination) {
-              // Keep cursor if there are less than value projects
-              if (projects.length < nextCursor.value) {
-                projectCursorsByOrganizationSlug[organization.slug] = currentCursor;
-              } else {
-                projectCursorsByOrganizationSlug[organization.slug] = nextCursor;
-              }
+            store.dispatch(LoadingAction(true, '${organization.name} -> Checking for sessions...', ++currentProgress / fullProgress));
+            try {
+              final projectsWithSessionsForOrganization = await api.projectIdsWithSessions(organization.slug);
+              projectIdsWithSessions.addAll(projectsWithSessionsForOrganization);
+            } catch (e) {
+              Sentry.addBreadcrumb(Breadcrumb(message: 'Org has no projects -> $e', level: SentryLevel.error));
```

Comment:
Maybe we should do a fallback here, and just show the projects we have instead of showing none? User will have projects without session data, but i think this might be better than showing none?

Reply:
I think showing the empty screen that says, no projects with session data makes more sense. Otherwise, the logic is different here for no reason.